### PR TITLE
[Vulkan] Use reference counting based wrapper layer

### PR DESCRIPTION
**Pull request by bobcao3**
Wraps the core vulkan object creation API in shared ptrs and provide references to keep track of object lifetime. This is a much cleaner approach than trying to track them in the RHI.

And renamed `vulkan_api.cpp` to `embedded_device.cpp`, so that the light resource wrapper can be called `vulkan_api.cpp`

### CODE REVIEW

**Comment by bobcao3** [comment]
+ It might be useful to put this thin wrapper into another repo so we can use it in any other vulkan projects.

File: imgui.ini [hunk 1]
```diff
@@ -0,0 +1,24 @@
+[Window][Debug##Default]
```
Comment (AmesingFlank) [inline imgui.ini:1]: remove this lol. I apologize that GGUI keeps generating this... Maybe we could put it in a gitignore.

**Comment by AmesingFlank** [comment]
Otherwise lgtm.

**Review by AmesingFlank — APPROVED** [review]

**Pull request merged**